satpy/composites/__init__.py

```diff
@@ -410,17 +424,24 @@ def __call__(self, projectables, optional_datasets=None, **info):
         # Check if the sun-zenith angle was provided:
         if sun_zenith is None:
             from pyorbital.astronomy import sun_zenith_angle as sza
-            lons, lats = nir.info["area"].get_lonlats()
-            sun_zenith = sza(nir.info['start_time'], lons, lats)
+            lons, lats = self._nir.info["area"].get_lonlats()
+            sun_zenith = sza(self._nir.info['start_time'], lons, lats)
 
-        refl39 = Calculator(nir.info['platform_name'],
-                            nir.info['sensor'], nir.id.wavelength[1])
+        return self._refl3x.reflectance_from_tbs(sun_zenith, self._nir, self._tb11, tb_ir_co2=tb13_4)
 
-        proj = Dataset(refl39.reflectance_from_tbs(sun_zenith, nir,
-                                                   tb11, tb13_4) * 100,
-                       **nir.info)
-        proj.info['units'] = '%'
-        self.apply_modifier_info(nir, proj)
+
+class NIREmissivePartFromReflectance(NIRReflectance):
+
+    def __call__(self, projectables, optional_datasets=None, **info):
+        """Get the emissive part an NIR channel after having derived the reflectance. 
+        Not supposed to be used for wavelength outside [3, 4] µm.
+        """
+        self._init_refl3x(projectables, optional_datasets)
+        _ = self._get_reflectance(optional_datasets)
```

Comment:
This isn't required anymore.

Reply:
How do you mean!?
The `__call__` method of the super class is not called, so I need this here as well, right?
I have done as you suggested, but the code increased by three lines.
I think the main problem with our concept of reusing code is that we have no `__init__`.

Reply:
The following actually works, but is it really better!?
```
    def __call__(self, projectables, optional_datasets=None, **info):
        """Get the emissive part an NIR channel after having derived the reflectance. 
        Not supposed to be used for wavelength outside [3, 4] µm.
        """

        _ = super(NIREmissivePartFromReflectance, self).__call__(projectables, optional_datasets, **info)
        del(_)
        # self._init_refl3x(projectables)
        #_ = self._get_reflectance(projectables, optional_datasets)
        _nir, _ = projectables
        proj = Dataset(self._refl3x.emissive_part_3x(), **_nir.info)

        proj.info['units'] = 'K'
        self.apply_modifier_info(_nir, proj)

        return proj
```